controllers/node.go

```diff
@@ -538,6 +538,10 @@ func createNode(w http.ResponseWriter, r *http.Request) {
 		logic.ReturnErrorResponse(w, r, logic.FormatError(err, "badrequest"))
 		return
 	}
+	if !validName(node.Name) {
```

Comment:
node.Name is already validated in logic.CreateNode and logic.Update node...  This validation is redundant

Reply:
@mattkasun  this Pr is as a result of this issue https://github.com/gravitl/netmaker/issues/1764

Reply:
the resolution we agreed on https://toil.kitemaker.co/cQARgD-Gravitl/Nk7Fgo-Community/items/753

Reply:
try and change a node name to something other than char or - on existing server .... It will fail ... the nodes checks are not needed

Reply:
@mattkasun removed the redundancy